src/reflect/scala/reflect/internal/ReificationSupport.scala

```diff
@@ -351,9 +351,13 @@ trait ReificationSupport { self: SymbolTable =>
                                        List[Tree], List[Tree], ValDef, List[Tree])] = tree match {
         case ClassDef(mods, name, tparams, impl) =>
           val X = if (mods.isCase) UnMkTemplate.asCase else UnMkTemplate
-          val X(parents, selfType, ctorMods, vparamss, earlyDefs, body) = impl
-          if (ctorMods.isTrait || ctorMods.hasFlag(JAVA)) None
-          else Some((mods, name, tparams, ctorMods, vparamss, earlyDefs, parents, selfType, body))
+          impl match {
```

Comment:
Definitely use Frank Morgan's law here. The one about not looking behind the curtain. BTW, his real name was Wuppermann. I definitely want the t-shirt with the big W on it. Sorry I didn't arrive in time for this review.